### PR TITLE
Add gql

### PR DESCRIPTION
**Pull request by anarellano**

### CODE REVIEW

**Comment by anarellano** [comment]
Added i on table.js .map() because for a unique idetifer. Page kept messing up because there were duplicate names

File: client/pages/table.tsx [hunk 1]
```diff
@@ -3,120 +3,98 @@
...
-    try {
-      const url = new URL("/getPlayers", "http://localhost:8000");
-
-      url.searchParams.append("page", page.toString());
-      url.searchParams.append("limit", limit.toString());
 
-      if (nbaPlayer) url.searchParams.append("nbaPlayer", nbaPlayer);
-      if (team) url.searchParams.append("team", team);
-      if (position) url.searchParams.append("position", position);
+  const { data } = useQuery(GET_PLAYERS_QUERY, {
```
Comment (evanugarte) [inline client/pages/table.tsx]: i believe we can also do

```suggestion
  const { data, loading, error } = useQuery(GET_PLAYERS_QUERY, {
```

lets add some logic where we show some error information if the method call failed. feel we can also do this in another pr, up to you

File: client/pages/table.tsx [hunk 2]
```diff
@@ -3,120 +3,98 @@
...
-  blocks: number;
-  turnovers: number;
-  personalFouls: number;
-  pointsPerGame: number;
-}
+const GET_PLAYERS_QUERY = gql`
+  query GetAllPlayers(
+    $team: String
+    $position: String
+    $player: String
```
Comment (evanugarte) [inline client/pages/table.tsx]: lets call this `search` instead, reads better

File: client/pages/table.tsx [hunk 3]
```diff
@@ -178,10 +161,10 @@ const table = () => {
         <div>
           <input
             type="text"
-            value={nbaPlayer}
-            onChange={(e) => setNbaPlayer(e.target.value)}
-            onKeyDown={handleKeyDown}
-          ></input>
+            value={SearchPlayer}
+            onChange={(e) => setSearchPlayer(e.target.value)}
+            // onKeyDown={handleKeyDown}
```
Comment (evanugarte) [inline client/pages/table.tsx]: why was this removed? do we call graphql every time the input changes?

File: client/pages/table.tsx [hunk 4]
```diff
@@ -192,6 +175,7 @@ const table = () => {
         <div>
           <PositionDropDown onClick={handlePositionClick} />
         </div>
+        <div><input placeholder="limit per page"type="text" onChange={e=>(setLimit(Number(e.target.value)))}/></div>
```
Comment (evanugarte) [inline client/pages/table.tsx]: do we need this, seems like it adds a lot of complication

File: client/pages/table.tsx [hunk 5]
```diff
@@ -229,8 +213,9 @@ const table = () => {
           </tr>
         </thead>
         <tbody>
-          {players.map((player) => (
-            <tr key={player.player}>
+          {/* added i because there are duplicates */}
+          {players.map((player, i) => (
+            <tr key={i}>
```
Comment (evanugarte) [inline client/pages/table.tsx]: we still shouldnt use indexes, lets do something like
`player.player + player.team`, would that be unique even if there are duplicate names?

File: client/pages/table.tsx (hunk 4 above)
Comment (anarellano) [reply client/pages/table.tsx]: Deleted

File: client/pages/table.tsx (hunk 5 above)
Comment (anarellano) [reply client/pages/table.tsx]: changed the index, but a lot of errors happening again on the console,

File: client/pages/table.tsx (hunk 1 above)
Comment (anarellano) [reply client/pages/table.tsx]: added error and loading headers

File: client/pages/table.tsx (hunk 2 above)
Comment (anarellano) [reply client/pages/table.tsx]: done ty

File: client/pages/table.tsx (hunk 5 above)
Comment (anarellano) [reply client/pages/table.tsx]: change might be too crazy, please check for refrence

File: client/pages/table.tsx [hunk 6]
```diff
@@ -229,8 +217,9 @@ const table = () => {
           </tr>
         </thead>
         <tbody>
+          {/* added i because there are duplicates */}
           {players.map((player) => (
-            <tr key={player.player}>
+            <tr key={player.player && player.team && player.pointsPerGame}>
```
Comment (evanugarte) [inline client/pages/table.tsx]:
```suggestion
            <tr key={player.player + player.team + player.pointsPerGame}>
```

File: client/pages/table.tsx [hunk 7]
```diff
@@ -136,56 +105,75 @@ const table = () => {
...
     setPosition(selectedPosition);
     setPage(0);
-    fetchPlayerData(page, limit, nbaPlayer, team, position);
   };
 
+  if (loading) {
+    return <h1> DATA IS LOADING...</h1>;
+  }
+  if (error) {
+    return <h1> Could Not Grab Data...</h1>;
```
Comment (evanugarte) [inline client/pages/table.tsx:169]:
```suggestion
    return <h1> Could Not Grab Data {error}</h1>;
```
